packages/preview-middleware-client/src/adp/controllers/ControllerExtension.controller.ts

```diff
@@ -120,28 +134,108 @@ export default class ControllerExtension extends BaseDialog {
      */
     async onCreateBtnPress(event: Event) {
         const source = event.getSource<Button>();
-        source.setEnabled(false);
+        const controllerExists = this.model.getProperty('/controllerExists');
 
-        const controllerName = this.model.getProperty('/newControllerName');
-        const viewId = this.model.getProperty('/viewId');
+        if (!controllerExists) {
+            source.setEnabled(false);
 
-        await this.createNewController(controllerName, viewId);
+            const controllerName = this.model.getProperty('/newControllerName');
+            const viewId = this.model.getProperty('/viewId');
+
+            await this.createNewController(controllerName, viewId);
+        } else {
+            const controllerPath = this.model.getProperty('/controllerPath');
+            window.open(`vscode://file/${controllerPath}`);
```

Comment:
Please change to `vscode://file${controllerPath}`, otherwise opening in VSCode will not work, because of `//` after `file`.

Reply:
Opening in VSCode on Mac? Otherwise, on Windows, it works. And you mean those `//` should also be platform-specific?

Reply:
On MAC the current implementation does NOT work. I needed to remove `/` after `file` to make it work. Does it work also without the `/` on Windows?

Reply:
I will test on both, since I still have both laptops :)

Reply:
Update: it does not work on Windows without slash `/`. For Windows the controllerPath is `C:\\Users` and for Mac it's`/Users/..`. So probably I need to add a slash when the OS is Windows, otherwise no slash.

Reply:
@nikmace Can it be sorted out by just using `path.join`?

Reply:
We can use path.join on the server only. Just tested the last commit on Windows, doesn't work..

Reply:
What URL is working for on Windows?

Reply:
For windows you need a slash after `file` like this: `"file/C:\\Users\\...\\open-ux-tools-playground\\apps\\adp.v2app\\webapp\\changes\\coding\\share.js"`. And for Mac you need to remove the slash after `file`

Reply:
Update 2: the last commit works ob both systems

Reply:
@nikmace We can also use `/` on Windows. Hence a path like `vscode://file/C:/Users/git/open-ux-tools/package.json` will work. I just tested it on windows. So I would propose to convert the Windows paths to POSIX as done in here https://stackoverflow.com/a/63251716

Reply:
Again the same problem: For Windows the controllerPath is `C:/Users/..` and for Mac it's `/Users/..`. So for Windows I need to add a slash after file and for Mac no

Reply:
Thx, tested it works!